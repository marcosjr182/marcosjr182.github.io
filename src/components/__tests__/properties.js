import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Properties from '../properties';

const items = { Water: 5, Food: 4, Ammunition: 3, Medication: 2 }
let wrapper = shallow(<Properties items={items} columns='3' />)

describe('Properties component', () => {

  it('should render the properties correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the four types of items', () =>{
    expect(wrapper.find('Property').length).toBe(4);
  })

  it('should not break if items is empty', () =>{
    wrapper = shallow(<Properties items={undefined} columns='3' /> )
    expect(wrapper.find('Property').length).toBe(0);
  })

})
