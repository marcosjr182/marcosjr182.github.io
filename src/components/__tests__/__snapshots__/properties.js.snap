exports[`Properties component should render the properties correctly 1`] = `
<div
  className="item-list">
  <Property
    columns="3"
    name="Water"
    value={5} />
  <Property
    columns="3"
    name="Food"
    value={4} />
  <Property
    columns="3"
    name="Ammunition"
    value={3} />
  <Property
    columns="3"
    name="Medication"
    value={2} />
</div>
`;
